### ISSUE TITLE
spring-cloud-starter-kubernetes-client-config not working on minikube

### ISSUE THREAD

**Issue by sunnyramsey** [spring-cloud/spring-cloud-kubernetes#885]
When I using spring-cloud-starter-kubernetes-client-config for configmap watching, I encountered some problems.

First exception is "PKIX path building failed: sun.security.provider.certpath.SunCertPathBuilderException: unable to find valid certification path to requested target", then I added the ca certificate file to the jre, solved it.

Second exception is the watch api return 403 because of service account when I start application using IDEA outside of cluster,  application has already loaded the kubeconfig file which kubectl used, but still use anonymous user. When I deploy application to the cluster, no such problem.

But app still can't work, I got "extension (5) should not be presented in certificate_request" when watch request is sending.  I am using open-jdk-11, so I add -Djdk.tls.client.protocols=TLSv1.2 to the JVM args, it works. But another exception arises:

2021-10-14 15:05:05.352 ERROR 1 --- [s.V1ConfigMap-1] i.k.c.informer.cache.ReflectorRunnable   : class io.kubernetes.client.openapi.models.V1ConfigMap#Reflector loop failed unexpectedly

io.kubernetes.client.openapi.ApiException:
	at io.kubernetes.client.util.Watch.createWatch(Watch.java:105) ~[client-java-13.0.0.jar!/:na]
	at io.kubernetes.client.informer.SharedInformerFactory$1.watch(SharedInformerFactory.java:215) ~[client-java-13.0.0.jar!/:na]
	at io.kubernetes.client.informer.SharedInformerFactory$1.watch(SharedInformerFactory.java:203) ~[client-java-13.0.0.jar!/:na]
	at io.kubernetes.client.informer.cache.ReflectorRunnable.run(ReflectorRunnable.java:120) ~[client-java-13.0.0.jar!/:na]
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515) ~[na:na]
	at java.base/java.util.concurrent.FutureTask.runAndReset(FutureTask.java:305) ~[na:na]
	at java.base/java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:305) ~[na:na]
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) ~[na:na]
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) ~[na:na]
	at java.base/java.lang.Thread.run(Thread.java:834) ~[na:na]

I still dont know what happend. But when I change spring-cloud-starter-kubernetes-client-config to spring-cloud-kubernetes-fabric8-config, everything works great! I dont need ca cert, I can even debug on the IDEA.

My spring-cloud-starter-kubernetes-client-config version is 2.0.4, spring cloud version is 2020.0.4, open jdk version is 11.0.6, minikube version is 1.22.2.

**Comment by kabennett** [comment]
Using Spring Boot 2.5.6, Spring Cloud AWS 2.3.2, and Spring Cloud 2020.0.3, our k8s pods start just fine in Minikube. After only upgrading Spring Cloud to 2020.0.4, I encounter this exact same error when the k8s pods start in Minikube.

**Comment by ryanjbaxter** [comment]
I am trying to track down the change but have been unsuccessful to this point.

@wind57 does anything come to mind that might have caused this?

**Comment by wind57** [comment]
I've looked at the PR, but I can't connect the dots, how that helps. Can you please briefly explain? Pretty please?

**Comment by ryanjbaxter** [comment]
I mean to add some comments to the PR, I will do that, but here is the gist

This broke as a side effect of this change
https://github.com/spring-cloud/spring-cloud-kubernetes/pull/868

The APIClient we configured within the cluster and then set as the default contains the Minikube CA.  In the `ChangeDetector` implementations we were creating a `SharedInformer` using the default constructor.  The default constructor pulls the default API client from `Configuration`.  In 2.0.3 this was the one we created, in 2.0.4 its not.  We need to pass an APIClient that was configured in the cluster so it has the CA cert.  I could get the ApiClient from the CoreApi instance but we would need to set the `readTimeout` to 0 and I didn't want to do that for fear of messing something else up, so I created a new instance.

Hope that makes sense.
